tapdb/sqlc/migrations/000007_universe.up.sql

```diff
@@ -81,3 +81,29 @@ CREATE VIEW universe_stats AS
     FROM universe_events u
     JOIN universe_roots roots ON u.universe_root_id = roots.id
     GROUP BY roots.asset_id, roots.group_key, roots.namespace_root;
+
+-- This table contains general configuration for universe federation syncing.
+CREATE TABLE IF NOT EXISTS federation_general_sync_config(
+    -- This ID column is used to ensure that only one row exists in this table.
+    id INTEGER PRIMARY KEY,
+
+    -- This flag indicates that, in general, only issuance proofs should be
+    -- synced.
+    only_sync_issuance_proofs BOOLEAN NOT NULL
+);
+
+-- This table contains universe (asset/asset group) specific federation sync
+-- configuration.
+CREATE TABLE IF NOT EXISTS federation_uni_sync_config (
```

Comment:
So is the idea that: 
  * We start blank with no sync config. 
  * This means sync everything, but general config says if it should be issuance or not. 
  * As you add a sync config, then we flip to not syncing everything, and only the items with a specified config. 

What about the case of you not necessarily syncing, but accepting insertions? Is this config used as well? So I only care about `A, B, C` and nothing else, someone tries to insert `D`, so then I reject that (with a predicable error).

Reply:
Just wanting to confirm we have the same mental model here.

Reply:
> As you add a sync config, then we flip to not syncing everything, and only the items with a specified config.

I have a different thought on this point. In my mind, a universe specific config should supersede the general config.

So if we only wanted to sync those universes which had specific configuration settings, then the general config should be set to "no syncing". And then the universe specific configs would supersede that general stance.

> What about the case of you not necessarily syncing, but accepting insertions? Is this config used as well? So I only care about A, B, C and nothing else, someone tries to insert D, so then I reject that (with a predicable error).

Where syncing is concerned, in its current state, this PR captures that behaviour with a general "no sync" config and then universe specific configs for `A`, `B`, `C`.

We could conceivably reuse these tables as permissions around universe proof couriering (universe specific permissions on `InsertProof`). I guess I can name the tables in a more general way in case we want to go down that route. What do you think @Roasbeef ?

Reply:
>  then the general config should be set to "no syncing". And then the universe specific configs would supersede that general stance.

I think that makes sense. 

> We could conceivably reuse these tables as permissions around universe proof couriering (universe specific permissions on InsertProof). I guess I can name the tables in a more general way in case we want to go down that route. 

I'm down with that approach. Seems like just a bit more logic to give us fine grained control from day one. Then as above, the specific config supersedes the general. 

```sql
-- This table contains universe (asset/asset group) specific remote proof
-- insert/export configuration.
CREATE TABLE IF NOT EXISTS uni_remote_permissions (
    -- This field contains the byte serialized ID of the asset that this
    -- configuration applies to.
    asset_id  BLOB CHECK(length(asset_id) = 32),

    -- This field contains the byte serialized group key public key of the asset
    -- group that this configuration applies to.
    group_key BLOB CHECK(LENGTH(group_key) = 32),

    -- This field is an enum representing the proof type stored in the given
    -- universe.
    proof_type TEXT NOT NULL CHECK(proof_type IN ('issuance', 'transfer')),

    -- This field is a boolean that indicates whether or not the given universe
    -- should accept remote proof insertion (via sync or proof courier).
    allow_remote_insert BOOLEAN NOT NULL,

    -- This field is a boolean that indicates whether or not the given universe
    -- should accept remote proof export (via sync or proof courier).
    allow_remote_export BOOLEAN NOT NULL,

    -- Both the asset ID and group key cannot be null at the same time.
    CHECK (asset_id IS NOT NULL OR group_key IS NOT NULL),

    -- The primary key is the universe identifier: the asset ID and group key.
    PRIMARY KEY (asset_id, group_key, proof_type)
);
```

IIUC, with this mode, if I want to allow all insertions, then we need mirrored bools of `allow_remote_insert` and `allow_remote_export` in the general config?